clang/test/SemaSYCL/kernelname-enum.cpp

```diff
@@ -1,13 +1,14 @@
 // RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -Wno-sycl-2017-compat -verify %s
 
+// expected-error@Inputs/sycl.hpp:220 2{{kernel name is invalid. Unscoped enum requires fixed underlying type}}
```

Comment:
This and in other places, what changed that we are now pointing into include file locations?
I think this will confuse users.

Reply:
We already do this for some diagnostics. The functionality was implemented here  - https://github.com/intel/llvm/pull/2474. See comment: https://github.com/intel/llvm/pull/2289#discussion_r469244709

Reply:
In this [PR](https://github.com/intel/llvm/pull/2289/files#diff-c88a590aca6d5f8e92dd1f8b6ec05a49R2023) , Kernel invocation function location was set to `Args[0]->getExprLoc()` . If the kernel invocation function(single_task, paralle_for) is defined in the test file, the diagnostics will point to the line number where the function was defined in the test file, but if the kernel function is defined in a header file, it will point to the line number in the header file.

Reply:
I see. The genie is already out of the bottle.